server/datastore/mysql/software.go

```diff
@@ -1238,58 +1262,106 @@ func (ds *Datastore) SyncHostsSoftware(ctx context.Context, updatedAt time.Time)
 	return nil
 }
 
-// HostsBySoftwareIDs returns a list of all hosts that have at least one of the specified Software
-// installed. It returns a minimal represention of matching hosts.
-func (ds *Datastore) HostsBySoftwareIDs(ctx context.Context, softwareIDs []uint) ([]*fleet.HostShort, error) {
-	queryStmt := `
-    SELECT 
-      h.id,
-      h.hostname,
-      if(h.computer_name = '', h.hostname, h.computer_name) display_name
-    FROM
-      hosts h
-    INNER JOIN
-      host_software hs
-    ON
-      h.id = hs.host_id
-    WHERE
-      hs.software_id IN (?)
-	GROUP BY h.id, h.hostname
-    ORDER BY
-      h.id`
-
-	stmt, args, err := sqlx.In(queryStmt, softwareIDs)
+func (ds *Datastore) HostVulnSummariesBySoftwareIDs(ctx context.Context, softwareIDs []uint) ([]fleet.HostVulnerabilitySummary, error) {
+	stmt := `
+		SELECT DISTINCT 
+			h.id,
+			h.hostname,
+			if(h.computer_name = '', h.hostname, h.computer_name) display_name,
+			COALESCE(hsip.installed_path, '') AS software_installed_path
+		FROM hosts h
+				INNER JOIN host_software hs ON h.id = hs.host_id AND hs.software_id IN (?)
+				LEFT JOIN host_software_installed_paths hsip ON hs.host_id = hsip.host_id AND hs.software_id = hsip.software_id
```

Comment:
```go
... AND hs.software_id = hsip.software_id
```
`getHostSoftwareInstalledPaths` joins with `s.id = hsip.software_id`, which one should it be?